libeos-payg/boottime-source.c

```diff
@@ -88,18 +89,21 @@ static const GSourceFuncs epg_boottime_source_funcs = {
 /**
  * epg_boottime_source_new:
  * @interval_ms: the timeout interval, in ms
+ * @error: return location for a #GError, or %NULL
  *
  * Like g_timeout_source_new(), but uses `CLOCK_BOOTTIME` to account for time
  * when the system is suspended.
  *
  * If @interval_ms is set to zero, the #GSource will be ready the next time
  * it's checked.
  *
- * Returns: (transfer full): a new `CLOCK_BOOTTIME` #GSource
+ * Returns: (transfer full) (nullable): a new `CLOCK_BOOTTIME` #GSource, or
+ *   %NULL with @error set
```

Comment:
Don’t add the `(nullable)` annotation for the case where a function can return `NULL` iff it returns an error. The existence of a `GError**` argument implies that. Only add `(nullable)` if it’s possible for a function to return `NULL` _without_ an error set.

Would also be useful to document the error code and the situation it can be returned in, i.e.
```
@error will be set to a #GIOError if, for example, the process runs out of file descriptors.
```